File: Java Source Files/ContactList.java
import java.util.Map.Entry;
import java.util.Set;

// ContactList extends AssociativeMemory
public class ContactList extends AssociativeMemory implements DictionaryInterface {

	protected boolean lock;

	// Default Constructor
	public ContactList() {
		lock = false;
	}

	// Sets lock to true
	public void setLock() {
		lock = true;
	}

	// Sets lock to false
	public void unsetLock() {
		lock = false;
	}

	// Prints the contents of the ContactList
	@Override
	public void print() {
		System.out.println(nameCode + " : " + valCode + " : Lock Status ("
				+ lock + ")");

		// Place map entries in a set. Sets are unordered.
		Set<Entry<String, String>> entries = super.map.entrySet();

		// Use iterator to print the set
		for (Entry<String, String> entry : entries)
			System.out.println(entry.getKey() + " : " + entry.getValue());
	}
}

File: C++ Source Files/ContactList.cpp
#include "ContactList.h"
#include <iostream>
#include <string>
using namespace std;

bool lock;

// Default Constructor calls on Dictionary and AssociativeMemory constructors
ContactList::ContactList() : Dictionary(), AssociativeMemory()
{
	lock = 0;	// default is unlocked
}

// Set lock
void ContactList::setLock()
{
	lock = 1;
}

// Unset lock
void ContactList::unsetLock()
{
	lock = 0;
}

// Prints contents of ContactList
void ContactList::print()
{
	cout << nameCode << " : " << valCode << " : Lock Status (" << lock << ")" << endl;
	map<string, string>::iterator p;
	for (p = map1.begin(); p != map1.end(); p++)
		cout << p->first << " : " << p->second << endl;
}